### ISSUE TITLE
Add deterministic VI methods

### ISSUE THREAD

**Issue by SamDuffield** [normal-computing/posteriors#76]
- DADVI ([paper](https://www.jmlr.org/papers/volume25/23-1015/23-1015.pdf)) pre-samples the Gaussian noise in the `init` and then is fully deterministic in `update` which makes for more stable training. More samples are needed so might need gradient accumulation support #52 
- Last-layer deterministic VI ([paper](https://arxiv.org/pdf/2404.11599.pdf)) provides a handy deterministic objective with linear last layers for regression and classification. Might be worth adding if we can generalise to exponential familiy losses and/or linearise the model.

**Comment by ifiaposto** [comment]
I am adding [SNGP](https://www.jmlr.org/papers/v24/22-0479.html) that is also a deterministic/ single-forward pass UQ method. not sure if it will work well on pre-trained models though, i.e., not necessarily regularized with spectral normalization.

**Comment by SamDuffield** [comment]
My understanding is that SNGP #68 training as in the [paper](https://www.jmlr.org/papers/v24/22-0479.html) is deterministic because they use a Laplace approximation. I imagine though you could do it with VI and even the deterministic VI methods above!

**Comment by SamDuffield** [comment]
Adding this which is definitely relevant
- DVI ([paper](https://arxiv.org/pdf/1810.03958)) deterministic VI although it seems to require quite intricate knowledge of the model
